bpf/k_unix_sock.h

```diff
@@ -0,0 +1,298 @@
+#ifndef _K_UNIX_SOCK
+#define _K_UNIX_SOCK
+
+#include "vmlinux.h"
+#include "bpf_helpers.h"
+#include "http_types.h"
+#include "tc_common.h"
+#include "bpf_dbg.h"
+#include "tracing.h"
+#include "k_tracer_defs.h"
+#include "k_send_receive.h"
+#include "protocol_http.h"
+#include "protocol_tcp.h"
+#include "k_unix_sock_tracker.h"
+
+static __always_inline struct unix_sock *unix_sock_from_socket(struct socket *sock) {
+    struct sock *sk;
+    BPF_CORE_READ_INTO(&sk, sock, sk);
+
+    if (!sk) {
+        return 0;
+    }
+
+    struct unix_sock *usock;
+    BPF_CORE_READ_INTO(&usock, sock, sk);
```

Comment:
non-rhetorical question: why is the double read required? Would something like the following not work?

```
const struct unix_sock *usock = BPF_CORE_READ(sock, sk);
```
?

Reply:
Ah thanks for catching it, it's a left over from some earlier changes.